### PR TITLE
Localize configMapGenerator, secretGenerator fields

### PR DESCRIPTION
**Pull request by annasong20**
This PR localizes the `configMapGenerator` and `secretGenerator` kustomization fields.

### CODE REVIEW

File: api/internal/localizer/util.go [hunk 1]
```diff
@@ -143,3 +144,53 @@ func locFilePath(fileURL string) string {
...
+	}
+	generator.EnvSources = locEnvs
+	generator.FileSources = locFiles
+	return nil
+}
+
+// parseFileSource parses the provided source, an entry under the `files` field for `configMapGenerator`, `secretGenerator`,
+// into the specified key, if one exists, and the file path. If only the file path is specified, parseFileSource returns
+// an empty key. parseFileSource returns an error for an invalid `files` entry.
+func parseFileSource(source string) (keyName, filePath string, err error) {
```
[the remark this answers is not in the document]
Comment (annasong20) [reply api/internal/localizer/util.go]: Done.

File: api/internal/localizer/util.go [hunk 2]
```diff
@@ -143,3 +144,53 @@ func locFilePath(fileURL string) string {
 	// so we can use it as is.
 	return filepath.Join(LocalizeDir, u.Hostname(), path)
 }
+
+// localizeGenerator localizes the file paths on generator.
+func localizeGenerator(lc *Localizer, generator *types.GeneratorArgs) error {
```
[the remark this answers is not in the document]
Comment (annasong20) [reply api/internal/localizer/util.go]: Done. 

I was originally trying to only define the absolute core methods on `Localizer`.

File: api/internal/generators/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		},
+		"missing key": {
+			Input: "=myfile",
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			key, file, err := ParseFileSource(test.Input)
+			if !test.Legal {
+				require.Error(t, err)
```
Comment (natasha41575) [inline api/internal/generators/utils_test.go:43]: nit: we should check the contents of the error message here too.

```
if test.expectedError != "" {
  require.Equal(t, err.Error(), test.expectedError)
}
```

**Review by natasha41575 — COMMENTED** [review]
/lgtm

Will defer to @KnVerey to make sure her comments are addressed for approval.

File: api/internal/generators/utils_test.go (hunk 3 above)
Comment (annasong20) [reply api/internal/generators/utils_test.go]: Done.

**Comment by KnVerey** [comment]
/approve
/lgtm
(but you need to rebase and resolve a conflict from your other PR FYI)

**Review by yuwenma — APPROVED** [review]

**Pull request merged**